packages/material-ui-lab/src/Timeline/Timeline.tsx

```diff
@@ -21,12 +22,42 @@ export const styles = () => ({
   alignAlternate: {},
 });
 
-const Timeline = React.forwardRef(function Timeline(props, ref) {
+export type TimelineClassKey = keyof WithStyles<typeof styles>['classes'];
+
+export interface TimelineProps extends StandardProps<React.HTMLAttributes<HTMLUListElement>> {
+  /**
+   * The position where the timeline's content should appear.
+   * @default 'left'
+   */
+  align?: 'left' | 'right' | 'alternate';
+  /**
+   * The content of the component.
+   */
+  children?: React.ReactNode;
+  /**
+   * Override or extend the styles applied to the component.
+   */
+  classes?: {
+    /** Styles applied to the root element. */
+    root?: string;
+    /** Styles applied to the root element if `align="left"`. */
+    alignLeft?: string;
+    /** Styles applied to the root element if `align="right"`. */
+    alignRight?: string;
+    /** Styles applied to the root element if `align="alternate"`. */
+    alignAlternate?: string;
+  };
+
+  ref?: React.Ref<HTMLUListElement>;
+}
+
+const Timeline = React.forwardRef<HTMLUListElement, TimelineProps>(function Timeline(props, ref) {
   const { align = 'left', classes, className, ...other } = props;
 
   return (
     <TimelineContext.Provider value={{ align }}>
       <ul
+        // @ts-expect-error unsafe string concat
         className={clsx(classes.root, classes[`align${capitalize(align)}`], className)}
```

Comment:
Could you move the `@ts-expect-error` to the exact argument? It's not clear what the issue is: `className` assignment, `clsx()` usage, `classes.root` read etc.

Reply:
The issue is with `classes[`align${capitalize(align)}`]`. Should I create a new variable?

Reply:
I don't think you need to. It should work with 

```tsx
className={clsx(
classes.root, 
// @ts-expect-error
classes[`align${capitalize(align)}`], 
className)}
```
and then let prettier give you the appropriate formatting.

Though can you leave a TODO comment to investigate this once 4.1 is released? This might work with [template literal types](https://devblogs.microsoft.com/typescript/announcing-typescript-4-1-beta/#template-literal-types).